lib/build.gradle.kts

```diff
@@ -12,22 +12,23 @@ repositories {
 }
 
 java {
-    toolchain {
-        languageVersion.set(JavaLanguageVersion.of(8))
-    }
-
     withSourcesJar()
     withJavadocJar()
 }
 
+tasks.withType<JavaCompile> {
```

Comment:
test should validate against the latest JDK, but compile against 8

Reply:
while I am not against it, what are the benefits of this change? We will see issues with the latest JDK early, but miss any potential issues with JDK 8, or am I missing something?

Reply:
Since the build is very cheap currently (with tests), I suggest we run the test suite with every LTS (8, 11, 17, 21) for every build instead of deciding if 8 or 21 is more important. What do you think?

Reply:
That would also be my preference, but lets add that after the other PRs are merged and keep it at 21 for now